see https://bitbucket.org/fenics-project/fiat/issues/27/sympy-12-breaks-diff-of-expression-without
and https://bitbucket.org/fenics-project/fiat/commits/8ceb29a

From 8ceb29a3b9b14668b9d387321a3f8f4126f9a159 Mon Sep 17 00:00:00 2001
From: Jan Blechta <blechta@karlin.mff.cuni.cz>
Date: Tue, 10 Jul 2018 23:30:03 +0200
Subject: [PATCH] Fix regression in SymPy 1.2

---
 FIAT/expansions.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/FIAT/expansions.py b/FIAT/expansions.py
index c90e2ac..df046e4 100644
--- a/FIAT/expansions.py
+++ b/FIAT/expansions.py
@@ -46,7 +46,7 @@ def _tabulate_dpts(tabulator, D, n, order, pts):
         out = []
         try:
             out = [sympy.diff(F, X[j]) for j in range(D)]
-        except AttributeError:
+        except (AttributeError, ValueError):
             # Intercept errors like
             #  AttributeError: 'list' object has no attribute
             #  'free_symbols'
-- 
2.10.5

